### PR TITLE
Add Azure IoT Hub integration documentation

### PR DESCRIPTION
**Pull request by adriansmares**
#### Summary

References https://github.com/TheThingsIndustries/lorawan-stack/issues/2808

#### Changes

- Add Azure IoT Hub guide
  - The guide is heavily tailored on the one we have for AWS IoT Core, minus the troubleshooting section. Hopefully I didn't carry over anything from there that's not supposed to be here.

Should we deleted the `microsoft-azure` integration guide, given that it is no longer that relevant ?

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links. Test with `HUGO_PARAMS_SEARCH_ENABLED=true` if style changes will affect the search bar.
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

**Review by nejraselimovic — COMMENTED** [review]
@adriansmares sorry for a bunch of suggestions, but I think these tweaks will improve the readability (especially for the first time readers). 

P.S. I added suggestions to place some images below their description (i.e. if the description says "select the **DIgital Twin** then the image with this button showed should go below), just because it's something we enforce though all our guides

I also think that we can remove the old Azure guide

**Review by neoaggelos — COMMENTED** [review]
Nothing to add

File: doc/content/integrations/cloud-integrations/azure-iot-hub/architecture.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+The key resources deployed in your Azure account are:
+
+- IoT Hub to manage your end devices and their Device Twins
+- IoT Hub Shared Access Policy that allows {{% tts %}} to register devices, submit traffic and update Device Twins
+- EventHub Namespace and EventHub for {{% tts %}} communication
+- Azure Function to submit events to {{% tts %}}
+- Storage Account used by the Azure Functions for trigger management
+
+This is a serverless deployment: there are no compute resources being deployed. The only continuous charges are by the Azure IoT Hub and the EventHub Namespace. All permissions are the minimum permissions for the integration to function.
```
[the remark this answers is not in the document]
Comment (adriansmares) [reply doc/content/integrations/cloud-integrations/azure-iot-hub/architecture.md:21]: The intent here is to say 'the integration has the minimum required access to your Azure resources'. It's similar to how we provide the minimum rights to API keys in TTS.

File: doc/content/integrations/cloud-integrations/azure-iot-hub/deployment.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+### Settings
+
+The parameters configure the integration:
+
+- **Location**: The region in which the Azure resources will be deployed. Defaults to the region used by the selected Resource Group.
+- **Name Suffix**: The suffix added to the Azure Resources. Defaults to a random string.
+- **Stack Cluster Address**: The address of your {{% tts %}} deployment. Can be retrieved from the **Azure IoT Hub** integration page in {{% tts %}}.
+- **Stack Application ID**: The application ID of your application. Can be retrieved from the **Azure IoT Hub** integration page in {{% tts %}}.
+- **Stack API Key**: The application API key that you generated before.
+- **Enable fallback route**: If the Azure IoT Hub fallback route should be added. If enabled, uplink messages will be submitted to the default `events` endpoint in your Azure IoT Hub.
```
[the remark this answers is not in the document]
Comment (adriansmares) [reply doc/content/integrations/cloud-integrations/azure-iot-hub/deployment.md:39]: This is more Azure specific behavior.

They have a system called routing, where you can choose where your messages go (i.e. some other queue in Azure). They also allow you to match which route to take based on the message (so you can say 'send to this queue only the messages that contain a temperature', for example).

If none of the routes match, the fallback route is taken, if enabled. The default behavior, when an Azure IoT Hub is created is to have a fallback route that sends any message to the `events` endpoint. This parameter configures if this default route should be added or not.

**Review by nejraselimovic — APPROVED** [review]
Great work, I think this is good to go!

**Pull request merged**